src/storage/core.rs

```diff
@@ -177,6 +261,10 @@ impl<K: Key> Storage<K> {
         meta: Option<Meta>,
     ) -> Result<()> {
         debug!("storage write with {:?}, {}b, {:?}", key, value.len(), meta);
+        // if active blob is set, this call will only check this fact and return
+        if self.create_active_blob().await? {

```

Comment:
Non-obvious behavior for a function with this name. It is better to rename function to `create_active_blob_if_not_exist`

Reply:
This behavior is defined by return type `Result<bool>`. If an error occurs, then `Result::Err(e)` is returned, otherwise `Result::Ok(true)` is returned in case, when active blob was really created, `Result::Ok(false)` - when it wasn't. I mentioned this in docs for functions, which behave like this. Should I rename them all? (Seems like now all functions behave like that and to have long names for most of public fucntions may be weird)

PS: I have also changed a comment a bit: this functions will not just return from function call, it will return `false` in case, when active blob already exists

Reply:
I think it's better to use `Error::ActiveBlobExists` instead of bool. And maybe rename to `try_create_active_blob()`